lte/gateway/c/session_manager/SessionState.cpp

```diff
@@ -1229,6 +1260,39 @@ bool SessionState::init_charging_credit(
   return true;
 }
 
+void SessionState::set_suspend_credit(
+    const CreditKey& charging_key, bool is_suspended,
+    SessionStateUpdateCriteria& update_criteria) {
+  auto it = credit_map_.find(charging_key);
+  if (it != credit_map_.end()) {
+    auto credit_uc = get_credit_uc(charging_key, update_criteria);
+    auto& grant    = it->second;
+    grant->set_is_suspended(is_suspended, credit_uc);
+  }
+}
+
+bool SessionState::is_credit_suspended(const CreditKey& charging_key) {
+  auto it = credit_map_.find(charging_key);
+  if (it != credit_map_.end()) {
+    auto& grant = it->second;
+    return grant->get_is_suspended();
+  }
+  return false;
+}
+
+void SessionState::get_rules_per_credit_key(
+    CreditKey charging_key, RulesToProcess& rulesToProcess) {
+  std::vector<std::string> static_rules_out;
+  std::vector<PolicyRule> dynamic_rules_out;
+
+  static_rules_.get_rule_ids_for_charging_key(charging_key, static_rules_out);
```

Comment:
I think you might be able to just do
`static_rules_.get_rule_ids_for_charging_key(charging_key, rulesToProcess.static_rules);` ? not 100% sure though :)

Reply:
oh thats right. I guess i got used to creating everything to avoid issues with references that are only visible at runtime